EIPS/eip-signTypedData.md

```diff
@@ -0,0 +1,439 @@
+# Structured data hashing and signing
+
+## Preamble
+
+    EIP:      <to be assigned>
```

Comment:
Please fix formatting per the new format, and rename and number this EIP to 712.